### PR TITLE
Various fixes.

### PR DESCRIPTION
**Pull request by ericpts**
* Now you can pass DEBUG=0 as an environment variable when calling make,
in order to run in release mode.
* Initialize run_was_called in Benchmark.
* Change asserts from !isnan() to isnormal(). The reasoning for this is
that some numbers end up being infinity, which is also incorrect
behaviour.

### CODE REVIEW

**Comment by mpeychev** [comment]
LGTM.

How do you pass the DEBUG=0 to the make?

**Comment by ericpts** [comment]
Like so:

```
DEBUG=0 make run
```

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,12 @@
 all: build/main
 $(shell mkdir -p build >/dev/null)
 
-OPTFLAGS := -DNDEBUG -flto -O2 -ftree-vectorize -march=native
+OPTFLAGS := -flto -ftree-vectorize -march=native
+
+ifeq ($(DEBUG), 0)
+	OPTFLAGS := $(OPTFLAGS) -DNDEBUG -O2
```
Comment (mpeychev) [inline Makefile]: General question: Is there any reason to not use `-O3` ?

**Comment by ericpts** [comment]
AFAIK `-O3` also enables some optimizations considered "unsafe", such as `-ffast-math`, which may mess up with the numerical stability of our code.

**Comment by mpeychev** [comment]
ok, better leave it with `-O2` for now but once we know the code is fine, can experiment with `-O3`, too

**Comment by ericpts** [comment]
Some results:

```
{
    "meta": {
        "ground_truth_implementation": "BaumWelchBlocking15",
        "repetitions": 3,
        "timestamp": "Mon Apr 29 10:10:15 2019\n"
    },
    "results": [
        {
            "configuration": {
                "nemissions": 120,
                "niters": 200,
                "nstates": 120,
                "ntime": 120
            },
            "implementations": [
                {
                    "cycle_counts": [
                        4427879722,
                        4446657214,
                        4471945278
                    ],
                    "implementation": "BaumWelchNaiveScaled",
                    "median_cycle_count": 4446657214,
                    "performance": 0.777876915969534
                },
                {
                    "cycle_counts": [
                        2002711842,
                        2030951886,
                        2048604450
                    ],
                    "implementation": "BaumWelchBlocking15",
                    "median_cycle_count": 2030951886,
                    "performance": 1.703118633111725
                },
                {
                    "cycle_counts": [
                        2277420352,
                        2283121708,
                        2295806672
                    ],
                    "implementation": "BaumWelchMemReorg",
                    "median_cycle_count": 2283121708,
                    "performance": 1.5150099041500595
                }
            ]
        }
    ]
}
```

**Pull request merged**